src/node_buffer.cc

```diff
@@ -460,10 +460,11 @@ void StringSlice<UCS2>(const FunctionCallbackInfo<Value>& args) {
   // need to reorder on BE platforms.  See http://nodejs.org/api/buffer.html
   // regarding Node's "ucs2" encoding specification.
   const bool aligned = (reinterpret_cast<uintptr_t>(data) % sizeof(*buf) == 0);
-  if (IsLittleEndian() && aligned) {
-    buf = reinterpret_cast<const uint16_t*>(data);
-  } else {
+  if (IsLittleEndian() && !aligned) {
```

Comment:
@bnoordhuis 
In this file, there are four cases that must be considered:
1. Little endian machine, properly aligned buf
2. Little endian machine, misaligned buf
3. Big endian machine, properly aligned buf
4. Big endian machine, misaligned buf

At present, reinterpreting "buf" as data is only done for the first case, and all other cases (2, 3 and 4) will involve a copy being made to produce an aligned version of unaligned data. This operation has the side effect of doing endian swapping ONLY when run on a big-endian system. This means that on a big-endian system, we will have data stored for the wrong endianness (little endian data gets swapped by alignment operation to big endian, then encode runs its own swapping function which swaps it to little endian again).

The proposed change will apply the alignment correction operation only for case 2 (little endian, misaligned). Case 1 (little endian, aligned) needs neither endianness swapping or alignment correction, while cases 3 and 4 will be handled using StringBytes::Encode's swap operation (which also takes care of alignment issues).

ALL cases need that buf is reinterpreted as a uint16_t variable, so we reinterpret either way.